packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx

```diff
@@ -38,7 +38,7 @@ export interface ConfirmModalProps {
   /** Confirm button variant */
   confirmButtonVariant?: ButtonVariant;
   /** Confirm action callback */
-  onConfirm(): void;
+  onConfirm(): void | Promise<void>;
```

Comment:
This is needed otherwise `await` won't work, right?

Reply:
yep exactly. we're basically allowing either a synchronous or asynchronous callback. if asynchronous, we disable the button so you can't spam click whilst it's in progress. if synchronous, `await`ing the callback still works it just resolves immediately 😄

Reply:
Also, how much of breaking change is it? Is it still backwards compatible?

Reply:
it should be completely backwards compatible 👍

Reply:
The small edge case is where previously if people were returning a Promise (which is valid when it can be `void`), the modal will now act a bit differently than before (it'll disable the submit button while the promise is pending), but I don't think this should result in any change in user-noticable behaviour (because the consumer still must manually stop rendering the modal)